doc/reference/peripherals/rtc.rst

```diff
@@ -0,0 +1,13 @@
+.. _rtc_interface:
+
+RTC
+###
+
+Overview
+********
+
```

Comment:
Missing an overview here...

Reply:
Yeah, I just copied counter.  The only peripherals that have overview are I2C and sensors.

Part of the reason I can't add one is that this is for RTC peripherals in general, and the PR provides a device-specific API.  I need instruction on how to document that.  From the PR description:
> A final question is where to collect the documentation. Because it's device specific, it really should be its own page at API Reference : Peripherals : RTC : DS3231. For now I've just put it under (newly created) RTC.

I've applied the other changes; thanks.